tests/app/dao/test_templates_dao.py

```diff
@@ -957,3 +959,67 @@ def test_template_with_provider_id_persists_provider_id(
     assert notify_db_session.session.get(Template, template.id).provider_id == provider.id
     # Teardown
     template_cleanup(notify_db_session.session, template.id)
+
+
+class TestDAOGetTemplateById:
+    def test_dao_get_template_by_id_with_no_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        result = dao_get_template_by_id(template.id)
+
+        assert result == template
+        assert isinstance(result, Template)
+        assert result.id == template.id
+        assert result.version == 1
+
+    def test_dao_get_template_by_id_with_specific_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        template.content = 'Updated content'
+        dao_update_template(template)
+
+        result = dao_get_template_by_id(template.id, version=1)
+
+        assert isinstance(result, TemplateHistoryData)
+        assert result.id == template.id
+        assert result.version == 1
+        assert result.content != 'Updated content'
+
+    def test_dao_get_template_by_id_with_version_none_returns_template_history_data(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        result = dao_get_template_by_id(template.id, version=99)
+
+        assert result is None
+
+    def test_dao_get_template_by_id_caching_behavior(
+        self,
+        notify_db_session: SQLAlchemy,
+        sample_template: Callable[..., Any],
+        mocker: MockerFixture,
+    ):
+        # Mock the cache to verify it's being used
+        db_spy = mocker.spy(notify_db_session.session, 'scalars')
+
+        template = sample_template()
+
+        # First call - should hit the database
+        result1 = dao_get_template_by_id(template.id, version=1)
+        assert db_spy.call_count == 1
+
+        db_spy.reset_mock()
+
+        # Second call - should use cached result
+        result2 = dao_get_template_by_id(template.id, version=1)
+        assert db_spy.call_count == 0
```

Comment:
Resetting the mock seems unnecessary.  You just need to ensure it doesn't increment again.
```suggestion
        # Second call - should use cached result
        result2 = dao_get_template_by_id(template.id, version=1)
        assert db_spy.call_count == 1
```